### PR TITLE
#2033 Newgrounds Audio Portal Connector

### PR DESCRIPTION
**Pull request by Ydeu**
Added a connector for [newgrounds](https://www.newgrounds.com/audio) audio portal.

**Additional context**
Not sure about the usage of document methods to get the names of artist and song in embedded player, but in the utils.js I have not found any function that could be of use and selectors did not pick up the data at all.

### CODE REVIEW

File: src/connectors/newgrounds.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+const artistSelector = '.ng-apg-media-artist';
+const trackSelector = '.ng-apg-media-title';
+
+function setupConnector() {
+	if (isEmbeddedPlayer()) {
+		setupEmbeddedPlayer();
+
+		Connector.getArtistTrack = () => {
```
Comment (alexesprit) [inline src/connectors/newgrounds.js]: I would move `Connector.getArtistTrack` inside of `setupXXXPlayer` function.

File: src/connectors/newgrounds.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+			let { artist, track } = Util.splitArtistTrack(document.getElementsByTagName('h2')[0].innerText, '-');
+
+			if (!artist) {
+				artist = document.getElementsByTagName('h4')[0].innerText;
+				track = document.getElementsByTagName('h2')[0].innerText;
```
Comment (alexesprit) [inline src/connectors/newgrounds.js]: 1. These selectors are ambiguous. I think they should be updated to make sure every selector is unique, e.g. `.item-user h4` for artist (username).
2. `Util.getTextFromSelectors` is prefferable to get text from elements, as it safer to use.

File: src/connectors/newgrounds.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+function isEmbeddedPlayer() {
+	return $('#audio-listen-player').length > 0;
+}
+
+function setupGlobalPlayer() {
+	Connector.playerSelector = '#_ngHiddenAudioPlayer';
+
+	Connector.playButtonSelector = '#global-audio-player-play';
```
Comment (alexesprit) [inline src/connectors/newgrounds.js]: You should use either play or pause button selector. Using both selectors is redundant, unless there's a special case to use both (but these cases should be documented).

**Comment by inverse** [comment]
Awesome :clap:

**Comment by nikos3194** [comment]
Thanks so much for including this site. :smiley:  :+1:

**Pull request merged**